independent-projects/arc/tests/src/test/java/io/quarkus/arc/test/cdi/bcextensions/CustomPseudoScopeTest.java

```diff
@@ -0,0 +1,155 @@
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Inherited;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import java.util.UUID;
+
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.enterprise.context.Dependent;
+import jakarta.enterprise.context.RequestScoped;
+import jakarta.enterprise.context.spi.AlterableContext;
+import jakarta.enterprise.context.spi.Contextual;
+import jakarta.enterprise.context.spi.CreationalContext;
+import jakarta.enterprise.inject.build.compatible.spi.BuildCompatibleExtension;
+import jakarta.enterprise.inject.build.compatible.spi.Discovery;
+import jakarta.enterprise.inject.build.compatible.spi.MetaAnnotations;
+import jakarta.inject.Inject;
+import jakarta.inject.Scope;
+import jakarta.inject.Singleton;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.arc.Arc;
+import io.quarkus.arc.ArcContainer;
+import io.quarkus.arc.test.ArcTestContainer;
+
+public class CustomPseudoScopeTest {
+    @RegisterExtension
+    public ArcTestContainer container = ArcTestContainer.builder()
+            .beanClasses(SingletonBean.class, ApplicationScopedBean.class, RequestScopedBean.class, DependentBean.class,
+                    MyService.class)
+            .buildCompatibleExtensions(new MyExtension())
+            .build();
+
+    @Test
+    public void test() {
+        ArcContainer container = Arc.container();
+        container.requestContext().activate();
+
+        MyService myService = container.select(MyService.class).get();
+
+        assertNotEquals(myService.get(), container.select(MyService.class).get().get());
+
+        SingletonBean singletonBean = Arc.container().select(SingletonBean.class).get();
+        assertEquals(singletonBean.get(), Arc.container().select(SingletonBean.class).get().get());
+        assertNotEquals(myService.get(), singletonBean.get());
+
+        ApplicationScopedBean applicationScopedBean = Arc.container().select(ApplicationScopedBean.class).get();
+        assertEquals(applicationScopedBean.get(), Arc.container().select(ApplicationScopedBean.class).get().get());
+        assertNotEquals(myService.get(), applicationScopedBean.get());
+
+        RequestScopedBean requestScopedBean = Arc.container().select(RequestScopedBean.class).get();
+        assertEquals(requestScopedBean.get(), Arc.container().select(RequestScopedBean.class).get().get());
+        assertNotEquals(myService.get(), requestScopedBean.get());
+
+        DependentBean dependentBean = Arc.container().select(DependentBean.class).get();
+        assertNotEquals(dependentBean.get(), Arc.container().select(DependentBean.class).get().get());
+        assertNotEquals(myService.get(), dependentBean.get());
+    }
+
+    @Singleton
+    static class SingletonBean {
+        @Inject
+        MyService service;
+
+        public String get() {
```

Comment:
I'd rename these methods to something like `getService()`... it will make the assertions above much clearer.

Reply:
Yeah, the number of `get()` calls in the test is a bit too high. I didn't bother too much, but I'll do something about it.

Reply:
Did some renaming here: instead of `MyService.get()`, there's `PrototypeBean.getId()`, and the other beans have `getPrototypeId()`. I also added javadoc to `@Prototype`.